options.go

```diff
@@ -268,7 +271,8 @@ func (o *Options) Validate() error {
 				return err
 			}
 			o.oidcVerifier = provider.Verifier(&oidc.Config{
-				ClientID: o.ClientID,
+				ClientID:        o.ClientID,
+				SkipIssuerCheck: o.InsecureOIDCSkipIssuerVerification,
```

Comment:
This verification skip doesn't work. The `oidc.NewProvider()` call above needs it as well, and since that's a thirdparty library, we can't change the signature. :(

Reply:
I've extracted this functionality into a separate PR (#467) because it's actually needed for multi-tenant Azure AADv2 OIDC support.